htlcswitch/circuit_map.go

```diff
@@ -305,15 +314,31 @@ func (cm *circuitMap) restoreMemState() error {
 
 // decodeCircuit reconstructs an in-memory payment circuit from a byte slice.
 // The byte slice is assumed to have been generated by the circuit's Encode
-// method.
-func decodeCircuit(v []byte) (*PaymentCircuit, error) {
+// method. If the decoding is successful, the onion obfuscator will be
+// reextracted, since it is not stored in plaintext on disk.
+func (cm *circuitMap) decodeCircuit(v []byte) (*PaymentCircuit, error) {
 	var circuit = &PaymentCircuit{}
 
 	circuitReader := bytes.NewReader(v)
 	if err := circuit.Decode(circuitReader); err != nil {
 		return nil, err
 	}
 
+	// If the error encrypter is nil, this is locally-source payment so
+	// there is no encrypter.
+	if circuit.ErrorEncrypter == nil {
```

Comment:
From just calling the above `Decode` method, I don't see how this could ever be non-nil.

Reply:
It can be nil if the payment is locally sourced, here's the place in `Decode` where it leaves it nil for this case https://github.com/lightningnetwork/lnd/blob/master/htlcswitch/circuit.go#L194

Reply:
right... I was looking at the wrong `Decode` method 😅

Reply:
No worries, it's a little funky cause it doesn't exactly follow just normal `Encode/Decode` flow :)